src/hotspot/share/opto/multnode.cpp

```diff
@@ -222,19 +222,15 @@ CallStaticJavaNode* ProjNode::is_uncommon_trap_if_pattern(Deoptimization::DeoptR
   if (in0->outcnt() < 2)  return nullptr;
   IfNode* iff = in0->as_If();
 
-  // we need "If(Conv2B(Opaque1(...)))" pattern for reason_predicate
-  if (reason != Deoptimization::Reason_none) {
-    if (iff->in(1)->Opcode() != Op_Conv2B ||
-       iff->in(1)->in(1)->Opcode() != Op_Opaque1) {
-      return nullptr;
-    }
+  // we need a ParsePredicate node for predicate reasons
+  if (reason != Deoptimization::Reason_none && !iff->is_ParsePredicate()) {
```

Comment:
Unrelated to your change but this code doesn't seem to do what the comment says. `reason != Deoptimization::Reason_none` is not "predicate reasons". I think this needs to be cleaned up at some point.

Reply:
Okay, I've missed one usage which is not `Deoptimization::Reason` - but let me still have a look if I can clean this method up somehow.